File: backend/src/main/java/com/example/backend/service/ProductService.java
package com.example.backend.service;

import com.example.backend.payload.ProductDto;
import com.example.backend.payload.ProductResponse;

public interface ProductService {
    ProductDto createProduct(ProductDto productDto);

    ProductResponse getAllProducts(int pageNo, int pageSize, String sortBy, String sortDir);

    ProductDto getProductById(long id);

    ProductDto updateProduct(ProductDto productDto, long id);

    void deleteProductById(long id);

    ProductResponse getProductsByNameAndCategory(String name, String category, int pageNo, int pageSize);

    ProductResponse getProductByProductCode(String productCode, int pageNo, int pageSize);
}

File: frontend/src/services/productService.js
import api from "./api";

const getAllProducts = () => {
	// return api.get(`/products?pageNo=${pageNo}&pageSize=${pageSize}`);
	return api.get(`/products`);
};

const getProductById = id => {
	return api.get(`/products/${id}`);
};

const createProduct = product => {
	return api.post("/products", product);
};

const updateProduct = product => {
	return api.put(`/products/${product.id}`, product);
};

const deleteProduct = id => {
	return api.delete(`/products/${id}`);
};

const getProductByProductCode = productCode => {
	return api.get(`/products/productCode/${productCode}`);
};

const getProductsByNameAndCategory = (name, category) => {
	return api.get(`/products/category?name=${name}&category=${category}`);
};

// eslint-disable-next-line import/no-anonymous-default-export
export default {
	getAllProducts,
	getProductById,
	createProduct,
	updateProduct,
	deleteProduct,
	getProductByProductCode,
	getProductsByNameAndCategory
};
